lib/mcb/courses_editor.rb

```diff
@@ -0,0 +1,73 @@
+module MCB
+  class CoursesEditor
+    def initialize(provider:, requester:, course_codes: [])
+      @cli = HighLine.new
+
+      @provider = provider
+      @requester = requester
+      @courses = course_codes.present? ? find_courses(course_codes) : provider.courses
+
+      check_authorisation
+    end
+
+    def run
+      finished = false
+      puts "Editing #{course_codes.join(', ')}"
+      until finished
+        choice = @cli.choose do |menu|
+          menu.choice(:exit) { finished = true }
+          menu.choice("edit title")
+        end
+
+        case choice
+        when "edit title"
+          edit_title
+        end
+      end
+    end
+
+  private
+
+    def edit_title
+      print_existing(:name)
+      update(name: ask_title)
+    end
+
+    def ask_title
+      @cli.ask("New course title?  ")
+    end
+
+    def check_authorisation
+      @courses.each { |course| raise Pundit::NotAuthorizedError unless can_update?(course) }
+    end
+
+    def print_existing(attribute_name)
```

Comment:
I expect these methods will see more use in future PRs with the rest of the fields :+1: